quodlibet/util/path.py

```diff
@@ -50,6 +53,38 @@ def fsn2glib(path):
     return path
 
 
+def uri2gsturi(uri):
+    """Takes a correct uri and returns a gstreamer-compatible uri"""
+    if not is_windows():
+        return uri
+    try:
+        fsn = uri2fsn(uri)
+    except ValueError:
+        return uri
+    try:
```

Comment:
Couldn't we just use the following, always?

```
from gi.repository import GLib
return GLib.filename_to_uri(uri2fsn(uri))
```

Reply:
GLib.filename_to_uri requires absolute paths, and I don't know if quodlibet ever uses relative paths

Reply:
Relative paths aren't valid in URIs, so you could do:

```
from gi.repository import GLib
try:
    return GLib.filename_to_uri(uri2fsn(uri))
except GLib.Error:
    return uri
```